Automatically set updates quotas

Bouncner opened the pull request:
When booting up, the merchant requests the marketplace quotas and sets its update interval accordingly.

Please be aware, that we only request the quota on bootup of the merchant. Later changed quotas might crash the merchant.

Bouncner committed:
Add get_update_rates() to receive quotas from marketplace

(cherry picked from commit dac732953db389b220372177d524e16b5a2a4d0e)

Bouncner committed:
Log API exceeded information

(cherry picked from commit e2ee7bc4e1d290886545b1d356127bab1bb8341a)

Bouncner committed:
Automatically request and set update freq. during startup

(cherry picked from commit 060c1ff42422f76f06eb1cad126063bc786453e2)

CarstenWalther commented on api/marketplace.py:
```diff
@@ -19,6 +19,10 @@ def get_offers(self, include_empty_offers: bool = False) -> List[Offer]:
 
+    def get_update_rates(self):
+        r = self.request('get', 'config')
+        return {key: value for (key, value) in r.json().items() if 'max' in key}
```
This could be more explicit. I prefer returning `max_updates_per_sale` and `max_req_per_sec` as tuple (or even as a dedicated object). Alternatively, there could be a function for each of the values.

(The merchant should not get infos about `consumer_per_minute`. We should remove this from this API call. But that's for another PR)

Bouncner replied:
As far as I have looked through the MP code, I don't see the `consumer_per_minute` used anyhow. So removing it from the API would definitely make sense, maybe even removing it from the marketplace?

The tuple does not make too much sense to me. I think your alternative of having two functions that return each one attribute at a time is the best solution. Will look at it next week.

Bouncner committed:
Ensure eventlet is available for flask server.

Bouncner committed:
Change function to get_request_limit()

Bouncner committed:
Add handling of API req limit violation using sleep()

Bouncner committed:
Adapt default update handling

Bouncner commented:
@CarstenWalther: addressed your concerns.

Also, I found crashing the merchants when API limit was exceeded problematic. As this can potentially happen due to several reasons -- which would otherwise force us to use extremely conservatice update rates -- I added a simple `sleep()` of which I am not sure whether I really like it.

What do you think?

CarstenWalther reviewed (approved):
I see two options:

1. Forward Timeout problem to merchant
- merchant has control how to react (retry/drop)
- requires much error handling

2. delayed retry (your implementation)
- delay might be unexpected

This implementation seems fine.

Bouncner commented:
I also thought about the forwarding the timeout handling, but I could neither think of a decent way to do that (well,  there is but that a rather large rewrite) nor do I have to time to fix that now.

CarstenWalther commented:
Whoops, I get an error here.
```
    return r['max_req_per_sec']
TypeError: 'Response' object is not subscriptable
```
Should be `return r.json()['max_req_per_sec']`
